pallets/api/src/fungibles/tests.rs

```diff
@@ -675,3 +675,95 @@ mod read_weights {
 		assert_eq!(token_exists.proof_size(), 3675);
 	}
 }
+
+mod ensure_codec_indexes {
+	use super::{Encode, RuntimeCall, *};
+	use crate::{fungibles::Call::*, mock::RuntimeCall::Fungibles};
+	#[test]
+	fn ensure_read_variant_indexes() {
+		let total_supply = TotalSupply::<Test>(Default::default());
+		let balance_of = BalanceOf::<Test> { token: Default::default(), owner: Default::default() };
+		let allowance = Allowance::<Test> {
+			token: Default::default(),
+			owner: Default::default(),
+			spender: Default::default(),
+		};
+		let token_name = TokenName::<Test>(Default::default());
+		let token_symbol = TokenSymbol::<Test>(Default::default());
+		let token_decimals = TokenDecimals::<Test>(Default::default());
+		let token_exists = TokenExists::<Test>(Default::default());
+		// Encode and assert on the first byte (the enum variant index)
+		assert_eq!(total_supply.encode()[0], 0, "TotalSupply variant index changed");
+		assert_eq!(balance_of.encode()[0], 1, "BalanceOf variant index changed");
+		assert_eq!(allowance.encode()[0], 2, "Allowance variant index changed");
+		assert_eq!(token_name.encode()[0], 8, "TokenName variant index changed");
+		assert_eq!(token_symbol.encode()[0], 9, "TokenSymbol variant index changed");
+		assert_eq!(token_decimals.encode()[0], 10, "TokenDecimals variant index changed");
+		assert_eq!(token_exists.encode()[0], 18, "TokenExists variant index changed");
+	}
+
+	#[test]
+	fn ensure_dispatchable_indexes() {
+		let transfer: RuntimeCall = Fungibles(transfer {
```

Comment:
As above, but here all the calls can be collected into an array as they are all of the same enum type -> fungibles::Call

Reply:
see latest commit